### ISSUE TITLE
`spread` pkg missing on some instances

### ISSUE THREAD

**Issue by upils** [canonical/github-runner-operator#210]
### Bug Description

While running tests in self-hosted runner I noticed the package [spread](https://github.com/snapcore/spread) was missing on some instances.

My job failed on:
- `two-xlarge-60-f76f155fd546fb6ed130eab3` (see https://github.com/canonical/ubuntu-image/actions/runs/7665245789/job/20890868438
- `two-xlarge-61-785cc4810cccfb8e44b4ca88` (see https://github.com/canonical/ubuntu-image/actions/runs/7665245789/job/20890996062)

And it succeeded (I least spread was found) on `ubuntu-image-agent-64-01` after a third retry on the same code.

AFAIK I never had to explicitly install `spread` on the runner so I thought it was installed by default.

I only notice the issue on these 2 particular instances by chance, but I guess other instances maybe affected.

### To Reproduce

If possible, create a github workflow with a constraint to run specifically on one of these two instances and trying to run spread.
 

### Environment

N/A

### Relevant log output

```shell
N/A
```

**Comment by yhaliaw** [comment]
The packages installed on the self-hosted runner provided by this charm is not the same as the GitHub-provided runners.

The image size of the GitHub-provided runners are quite large. Without any additional installation, the GitHub-provided runners are using 50+GiB of disk:
https://github.com/canonical/github-runner-operator/actions/runs/7691255463/job/20956277160#step:3:6

One of the planned feature is custom images where each job can specify a custom image to run on.
Currently, would installing `spread` package as part of the workflow be fine?
If not we can look at making it part of the image used by the charm.

**Comment by upils** [comment]
Yes, installing `spread` in the workflow is fine. We also even have runners dedicated to the project so this is not a big problem.

**Comment by cbartz** [comment]
It seems the user has found a solution (https://github.com/canonical/github-runner-operator/issues/210#issuecomment-1914124311). @jdkandersson do we have a feature request list for custom images somewhere already? Otherwise I'll close this issue.

**Comment by jdkandersson** [comment]
I think we can clos this issue for now as the solution of installing it on the runner is fine for now
